common/djangoapps/contentserver/middleware.py

```diff
@@ -36,14 +39,24 @@ def is_asset_request(self, request):
             request.path.startswith('/' + XASSET_LOCATION_TAG + '/')
             or
             request.path.startswith('/' + AssetLocator.CANONICAL_NAMESPACE)
+            or
+            self.is_versioned_request(request)
         )
 
+    def is_versioned_request(self, request):
+        return request.path.startswith(VERSIONED_ASSETS_PREFIX)
+
     def process_request(self, request):
         """Process the given request"""
         if self.is_asset_request(request):
             # Make sure we can convert this request into a location.
             if AssetLocator.CANONICAL_NAMESPACE in request.path:
                 request.path = request.path.replace('block/', 'block@', 1)
+
+            # If this is a versioned request, chop off the prefix.
+            if self.is_versioned_request(request):
+                request.path = re.sub(VERSIONED_ASSETS_PATTERN, '', request.path)
```

Comment:
Can we do some 303 redirection here if the hash doesn't match the current version we have? I really want to avoid the situation where we permanently cache in the CDN a URL value where the hash represents an older version of the content in that URL.